diff --git a/targets/json.js b/targets/json.js
index 70253723678448088ca104d26586eea855781f47..7b9bfa0d3833ff38be0618708ea520bfd1ce311d 100644
--- a/targets/json.js
+++ b/targets/json.js
@@ -4,5 +4,7 @@ module.exports = json_target;
 json_target.description = "JSON representation";
 
 function json_target(root, options, callback) {
-    callback(null, JSON.stringify(root, null, 2));
+    // Add hard code keepComments
+    // Based on the issue: https://github.com/protobufjs/protobuf.js/issues/1145
+    callback(null, JSON.stringify(root.toJSON({ keepComments: true }), null, 2));
 }
