diff --git a/src/enum.js b/src/enum.js
index 1c016209f4acb19ce330b62566fa92ea9a88aced..730b8e4dd146a3415e973f50c96aef5d604e7c52 100644
--- a/src/enum.js
+++ b/src/enum.js
@@ -87,7 +87,7 @@ function Enum(name, values, options, comment, comments, valuesOptions) {
  * @throws {TypeError} If arguments are invalid
  */
 Enum.fromJSON = function fromJSON(name, json) {
-    var enm = new Enum(name, json.values, json.options, json.comment, json.comments);
+    var enm = new Enum(name, json.values, json.options, json.comment, json.comments, json.valuesOptions);
     enm.reserved = json.reserved;
     return enm;
 };
